### PR TITLE
Add new authentication

### PR DESCRIPTION
**Pull request by EhsanMrh**

### CODE REVIEW

File: src/components/Pages/Authentication/Login.vue [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  data() {
+    return {
+      userName: "",
+      password: "",
+      form: this.$form.createForm(this, { name: "coordinated" })
+    };
+  },
+  methods: {
+    ...mapActions(["doLogin"]),
+    handleSubmit: function() {
```
Comment (pmzi) [inline src/components/Pages/Authentication/Login.vue]: handleSubmit(){}

File: src/components/Pages/Authentication/Login.vue [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      const { userName, password } = this;
+
+      this.form.validateFields((err, values) => {
+        if (!err) {
+          this.doLogin(userName, password)
```
Comment (pmzi) [inline src/components/Pages/Authentication/Login.vue]: return this.doLogin

File: src/components/Pages/Authentication/Login.vue [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    display: flex;
+    flex-direction: column;
+    max-width: 50rem;
+    justify-content: center;
+    padding: 0 5rem;
+    width: 30%;
+  }
+}
+
+.login-content__logo {
```
Comment (pmzi) [inline src/components/Pages/Authentication/Login.vue]: BEM

File: src/router/config.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  routes: [
+    {
+      name: "home",
+      path: "/",
+      component: Home
+    },
+    {
+      name: "login",
+      path: "/login",
+      component: Login
```
Comment (pmzi) [inline src/router/config.js]: lazy load
()=> import('....')

File: src/store/modules/login.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    doLogin({ commit }, userName, password) {
+      commit("LOGIN_START");
+      console.log("Doing Authenticat...!");
+      // Post data for authentication
+      axios({
+        url: "auth",
+        data: { userName, password },
+        method: "POST"
+      })
+        .then(response => {
```
Comment (pmzi) [inline src/store/modules/login.js]:
```javascript
.then(({ data: { token } })=>
...
```

File: src/stylesheets/abstract/_functions.scss [hunk 6]
```diff
@@ -0,0 +1,9 @@
+// for set font urls 
+@function font($font-name, $weight: '') {
+    $font-urls: url($public-path + 'fonts/' + $font-name + '/'+ $font-name + $weight + '.eot'),
```
Comment (pmzi) [inline src/stylesheets/abstract/_functions.scss]: Declare a function called font()

File: src/components/Pages/Authentication/Login.vue [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+            'doLogin'
+        ]),
+        handleSubmit: function () {
+            const {
+                userName,
+                password,
+            } = this;
+            
+            if(userName && password) {
+            this.doLogin({ userName, password})
```
[the remark this answers is not in the document]
Comment (EhsanMrh) [reply src/components/Pages/Authentication/Login.vue]: I try it but it doesn't work

File: src/router/config.js (hunk 4 above)
Comment (EhsanMrh) [reply src/router/config.js]: it will resolve in other branch

File: src/store/modules/login.js (hunk 5 above)
Comment (pmzi) [reply src/store/modules/login.js]: As I explained to you physically.

**Pull request merged**